cloud/amazon/rds_snapshot_copy.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: rds_snapshot_copy
+short_description: copies an existing rds snapshot
+description:
+    - copies an existing rds snapshot
+    - source snapshot must be in available state
+options:
+  source_region:
+    description:
+      - source region
+      - not provided then the environment default will be assumed
+    required: false
+  source_db_snapshot_identifier:
+    description:
+      - identifier of the source db snapshot
+      - must be in the "available" state
+      - if the source is in the same region then specify the snapshot id
+      - if the source is in a different region then specify the snapshot ARN
+    required: true
+  target_db_snapshot_identifier:
+    description:
+      - identifier for the copied snapshot
+      - cannot be empty
+      - must contain 1-255 alphanumeric characters or hyphens
+      - first character must be a letter
+      - cannot end with a hypen or contain two consecutive hyphens
+    required: true
+
+author: "Brock Haywood (@brockhaywood)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+
+'''
+
+try:
+    import boto.rds
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+
+def copy_snapshot(module, source_conn, conn,
+                  source_db_snapshot_identifier=None,
+                  target_db_snapshot_identifier=None):
+
+    if not source_db_snapshot_identifier or not target_db_snapshot_identifier:
+        module.fail_json(msg='Source DB Snapshot Id and Target DB Snapshot '
+                             'Id must be provided')
+
+    try:
+        snapshots = conn.get_all_dbsnapshots(
+            snapshot_id=source_db_snapshot_identifier)
```

Comment:
Ansible doesn't require strict PEP-8 linewidth compliance - not breaking here would improve readability.